### PR TITLE
adding way to detemrine inner and outer build

### PR DESCRIPTION
**Pull request by MIchaelRShea**
Fixes #

### CODE REVIEW

File: src/Build/Graph/ProjectInterpretation.cs [hunk 1]
```diff
@@ -126,6 +128,9 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
                         var platformPropertyInstance = ProjectPropertyInstance.Create(PlatformMetadataName, selectedPlatform);
                         referenceGlobalProperties[PlatformMetadataName] = platformPropertyInstance;
                     }
+                    // We will use this property to designate an inner build for platform negotiation 
+                    var platformInnerBuildPropertyInstance = ProjectPropertyInstance.Create("PlatformNegotiationInnerBuild", "true");
```
Comment (BenVillalobos) [inline src/Build/Graph/ProjectInterpretation.cs:132]: Doesn't this cause another evaluation? 

Say project A is x64;x86 and it builds both, AND project B references A and tells A to build x64. Wouldn't we get two evals of effectively the same instance? One with the inner build global prop/platform, and one with just the platform?

Comment (MIchaelRShea) [reply src/Build/Graph/ProjectInterpretation.cs:132]: this is a good point. in that cause we could identify a innerbuild by the lack of PlatformNegotiationouterBuild. that way we dont need to set anything extra and there is no dual eval

Comment (MIchaelRShea) [reply src/Build/Graph/ProjectInterpretation.cs]: pushed a change

File: src/Build/Graph/ProjectInterpretation.cs [hunk 2]
```diff
@@ -112,7 +112,9 @@ public IEnumerable<ReferenceInfo> GetReferences(ProjectInstance requesterInstanc
 
                     var  projectInstance = _projectInstanceFactory(
                         projectReferenceFullPath,
-                        null, // Platform negotiation requires an evaluation with no global properties first
+                        new Dictionary<string, string>(){
+                            {"PlatformNegotiationOuterBuild", "true"},
```
Comment (BenVillalobos) [inline src/Build/Graph/ProjectInterpretation.cs:116]: This doesn't feel right, maybe it's my understanding.

What exactly does `PlatformNegotiationOuterBuild` here mean? Is it similar to "outer builds" like building a `net472;net6.0` project and how it has individual "inner builds" of `net472` and `net6.0`? Or does "outer build" here mean something unique to platform negotiation?

**Pull request closed without merging**